apigee_m10n.module

```diff
@@ -36,23 +36,70 @@ function apigee_m10n_api_product_access(EntityInterface $entity, $operation, Acc
   return AccessResult::neutral();
 }
 
+/**
+ * Implements hook_theme_suggestions_HOOK().
+ */
+function apigee_m10n_theme_suggestions_rate_plan(array $variables) {
+  $suggestions = [];
+  $rate_plan = $variables['elements']['#rate_plan'];
+  $sanitized_view_mode = strtr($variables['elements']['#view_mode'], '.', '_');
+
+  $suggestions[] = 'apigee_rate_plan__' . $sanitized_view_mode;
+  $suggestions[] = 'apigee_rate_plan__' . $rate_plan->bundle();
+  $suggestions[] = 'apigee_rate_plan__' . $rate_plan->bundle() . '__' . $sanitized_view_mode;
+  $suggestions[] = 'apigee_rate_plan__' . $rate_plan->id();
+  $suggestions[] = 'apigee_rate_plan__' . $rate_plan->id() . '__' . $sanitized_view_mode;
+
+  return $suggestions;
+}
+
+/**
+ * Prepares variables for rate-plan templates.
+ *
+ * Default template: rate-plan.html.twig.
+ *
+ * @param array $variables
+ *   An associative array containing:
+ *   - elements: An array of elements to display in view mode.
+ *   - rate_plan: The Apigee  package rate plan entity.
+ *   - view_mode: View mode; e.g., 'full', 'teaser', etc.
+ *   - url: The canonical URL to view the plan directly.
+ *
+ * @throws \Drupal\Core\Entity\EntityMalformedException
+ */
+function template_preprocess_rate_plan(&$variables) {
+  $variables['view_mode'] = $variables['elements']['#view_mode'];
+
+  $variables['rate_plan'] = $variables['elements']['#rate_plan'];
+  /** @var \Drupal\apigee_m10n\Entity\RatePlanInterface $rate_plan */
```

Comment:
Note: given you require php 7.1 you could use assertions to verify these assumptions.  This validates its on development environment, but doesn't do anything on runtime. On top of  that IDE's are still able to reason  about it: `assert($rate_plan instanceof \Drupal\apigee_m10n\Entity\RatePlanInterface);`